src/OpenRiaServices.Tools/Framework/DataContractProxyGenerator.cs

```diff
@@ -399,7 +385,8 @@ protected virtual void GenerateProperty(PropertyDescriptor propertyDescriptor)
             property.Name = propertyName;
             property.Type = propTypeReference;
             property.Attributes = MemberAttributes.Public | MemberAttributes.Final; // final needed, else becomes virtual
-            List<Attribute> propertyAttributes = propertyDescriptor.ExplicitAttributes().Cast<Attribute>().ToList();
+            var propertyAttributes = propertyDescriptor.ExplicitAttributes().Cast<Attribute>()
+                .Where(a => a.GetType().Namespace != "System.Runtime.CompilerServices").ToList(); //Do not use attributes only used by the compiler
```

Comment:
Should RequiredAttribute be added if it contains RequiredMemberAttribute?

Reply:
Short answer: No. 

I thought about it to, butt they have different meaning.

* RequiredMemberAttribute ~= meember must be initialised when creating an instance (but null is a valid value for the member)
* RequiredAttribute ~= "null is not allow"

Maybe it would make sense in the future to infer Required from a based on "Not null" for reference types (maybe in combination med required member)